neuralop/training/algo.py

```diff
@@ -0,0 +1,163 @@
+import torch
+class Incremental():
+    def __init__(self, model, incremental, incremental_loss_gap,
+                incremental_resolution, incremental_eps, incremental_buffer,
+                incremental_max_iter, incremental_grad_max_iter, 
+                incremental_loss_eps, incremental_res_gap,
+                dataset_resolution, dataset_sublist, dataset_indices) -> None:
+        
+        if incremental and incremental_loss_gap:
+            raise ValueError(
+                "Incremental and incremental loss gap cannot be used together")
+            
+        self.model = model
+        self.ndim = len(model.n_modes)
+        self.incremental_grad = incremental
+        self.incremental_resolution = incremental_resolution
+        self.incremental_loss_gap = incremental_loss_gap
+
+        if self.incremental_grad:
+            # incremental gradient
+            self.buffer = incremental_buffer
+            self.grad_explained_ratio_threshold = incremental_eps
+            self.max_iter = incremental_max_iter
+            self.grad_max_iter =incremental_grad_max_iter
+
+        if incremental_loss_gap:
+            # incremental loss gap
+            self.eps = incremental_loss_eps
+            self.loss_list = []
+
+        if incremental_resolution:
+            # incremental resolution
+            self.epoch_gap = incremental_res_gap
+            self.indices = dataset_indices
+            self.resolution = dataset_resolution
+            self.sub_list = dataset_sublist
+
+            self.subsammpling_rate = 1
+            self.current_index = 0
+            self.current_logged_epoch = 0
+            self.current_sub = self.index_to_sub_from_table(self.current_index)
+            self.current_res = self.sub_to_res(self.current_sub)
+
+            print(f'Incre Res Update: change index to {self.current_index}')
```

Comment:
Can we add a `verbose` attribute and print only if it's >0 (and also print each update in the corresponding if so it only gets printed when it's actually updated?)

Reply:
Oh I mean initially we want it to print, so we can be sure of which resolution we are starting with before the 1st epoch. Secondly these statements are also only printed when it actually gets updated in the epoch_wise_res_increase function!

Reply:
Sure I can add that!